File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateActivity.java
package com.neyogiry.android.sample;

import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Toast;

import java.util.ArrayList;

public class DateActivity extends AppCompatActivity implements DateInterfaces.View{

    private FloatingActionButton fab;
    private RecyclerView rvDates;
    private DateAdapter adapter;

    private DatePresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_date);

        mPresenter = new DatePresenter(this);

        rvDates = (RecyclerView)findViewById(R.id.rv_dates);
        fab = (FloatingActionButton)findViewById(R.id.fab_add_date);

        setupRecyclerView();

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                addDate();
            }
        });
    }

    private void setupRecyclerView(){
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        rvDates.setLayoutManager(linearLayoutManager);
        adapter = new DateAdapter(new ArrayList<String>());
        rvDates.setAdapter(adapter);
    }

    public void addDate(){
        mPresenter.addDate();
    }

    @Override
    public void showDateList(ArrayList<String> dates) {
        adapter.updateList(dates);
    }

    @Override
    public void showError(String error) {
        Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
    }

}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateActivity.kt
package com.neyogiry.android.sample

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import android.support.v7.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_date.*


class DateActivity : AppCompatActivity(), DateInterfaces.View {

    private var adapter: DateAdapter? = null
    private var mPresenter: DatePresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_date)

        mPresenter = DatePresenter(this)

        setupRecyclerView()

        fab_add_date.setOnClickListener { mPresenter?.addDate() }
    }

    private fun setupRecyclerView() {
        val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        rv_dates.layoutManager = linearLayoutManager
        adapter = DateAdapter(ArrayList())
        rv_dates.adapter = adapter
    }

    override fun showDateList(dates: ArrayList<String>) {
        adapter?.updateList(dates)
    }

    override fun showError(error: String) {
        Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
    }

}
